sparta/test/SyncPort/SyncPort_test.cpp

```diff
@@ -425,6 +425,271 @@ TestSystem::~TestSystem()
+
+    static constexpr const char* name = "Source";
+
+private:
+    // Event set for the unit
+    sparta::EventSet ev_set;
+
+    // Self-scheduling event
+    sparta::UniqueEvent<> ev_do_work;
+
+    // Total commands that should be scheduled in both directions across
+    // the interfaces
+    static const uint64_t NUM_COMMANDS_TO_SEND = 10;
+};
+
+//////////////////////////////////////////////////////////////////////
+
+Source::Source(sparta::TreeNode* node, const Source::ParameterSet*) :
+    sparta::Resource(node),
+    ps(node),
+    out_data(&ps, "out_data", node->getClock()),
+    ev_set(node),
+    ev_do_work(&ev_set, "source_do_work_event", CREATE_SPARTA_HANDLER(Source, doWork))
+{
+
+}
+
+//////////////////////////////////////////////////////////////////////
+
+Source::~Source()
+{
+    std::cout << "Destructing '" << getName() << "'\n";
+}
+
+//////////////////////////////////////////////////////////////////////
+
+void Source::schedule_commands()
+{
+    // not driven in this cycle
+    EXPECT_FALSE(out_data.isDriven());
+
+    out_data.send('x');
+    EXPECT_TRUE(out_data.isDriven());
+
+    EXPECT_TRUE(out_data.isDriven(getClock()->currentCycle()));
+
+    sparta::Clock::Cycle clk_gap = out_data.computeNextAvailableCycleForSend(0,1);
+
+    for(uint32_t idx=1; idx<=NUM_COMMANDS_TO_SEND; idx++)
+    {
+        sparta::Clock::Cycle delay_cycles = idx * clk_gap;
+        auto driven_cycles = delay_cycles;
+
+        EXPECT_FALSE(out_data.isDriven(delay_cycles));
+
+        while(out_data.isDriven(driven_cycles)) driven_cycles++;
+
+        // send after delay (source)cycles
+        out_data.send('y', delay_cycles);
+
+        EXPECT_TRUE(out_data.isDriven(delay_cycles));
+
+        // trigger event in delay cycles
+        ev_do_work.schedule(delay_cycles);
+    }
+}
+
+void Source::doWork()
+{
+    //std::cout << __func__ << " isDriven=" << EXPECT_TRUE(out_data.isDriven())
+    //          << " source cycle=" << getClock()->currentCycle()
+    //          << std::endl;
+}
+//////////////////////////////////////////////////////////////////////
+//
+// This destination class for checking for SyncPort isDriven
+
+class Destn : public sparta::Resource
+{
+public:
+    class ParameterSet : public sparta::ParameterSet
+    {
+    public:
+        ParameterSet(sparta::TreeNode* tn) : sparta::ParameterSet(tn) {}
+    };
+
+    Destn(sparta::TreeNode* node, const Destn::ParameterSet*);
+    ~Destn();
+
+    // Callback for data.  This method is to test that the command is
+    // received before the data.  The data is ignored.
+    void data_callback(const char &);
+
+    void doWork() {};
+
+    sparta::PortSet ps;
+
+    // These are the classes we're actually testing
+    sparta::SyncInPort<char>     in_data;
+
+    static constexpr const char* name = "Destn";
+
+private:
+    // Event set for the unit
+    sparta::EventSet ev_set;
+
+    // Self-scheduling event
+    sparta::UniqueEvent<> ev_do_work;
```

Comment:
Nit: member naming convention does not match Sparta standards (missing trailing `_`)